src/native/http_request_utils.c

```diff
@@ -282,7 +290,12 @@ int aws_apply_java_http_request_changes_to_native_request(
     struct aws_byte_cursor marshalled_cur =
         aws_byte_cursor_from_array((uint8_t *)marshalled_request_data, marshalled_request_length);
 
-    result = s_unmarshal_http_request(message, &marshalled_cur);
+    enum aws_http_version version = s_unmarshal_http_request_to_get_version(&marshalled_cur);
+    if (version != aws_http_message_get_protocol_version(message)) {
+        result = aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    } else {
+        result = s_unmarshal_http_request(message, &marshalled_cur);
+    }
     (*env)->ReleasePrimitiveArrayCritical(env, marshalled_request, marshalled_request_data, 0);
 
     if (jni_body_stream) {
```

Comment:
look into this.
It's a little weird that we continue to do stuff after `result` is AWS_OP_ERR